src/resources/application.rs

```diff
@@ -648,34 +568,24 @@ pub async fn patch_application(
             Ok(data) => data,
             Err(resp) => return resp,
         };
-
-    // Fist, check if the model exists.
-    // TODO(joonas): we should likely fetch the version of the manifest that's running in Kubernetes
-    // TODO(joonas): Should this use model.status instead of model.get?
-    let get =
-        match wash_lib::app::get_model_details(&nats_client, Some(lattice_id.clone()), &name, None)
-            .await
-        {
-            Ok(res) => res,
-            Err(e) => {
-                return internal_error(anyhow!("unable to find app: {}", e));
+    let wadm_client = WadmClient::from_nats_client(&lattice_id, None, nats_client);
+    let current_manifest = match wadm_client.get_manifest(&name, None).await {
+        Ok(m) => m,
+        Err(e) => match e {
+            ClientError::ManifestLoad(_)
```

Comment:
Nit: You could just shorten this to `_ => return internal_error(anyhow!("unable to request app from wadm: {}", e))` after the `ClientError::NotFound` branch

Reply:
Good call, that would read much nicer, thanks for the reminder!